src/php7/php_solr.c

```diff
@@ -180,6 +180,12 @@ ZEND_BEGIN_ARG_INFO(SolrDocument_unserialize_args, 0)
 ZEND_ARG_INFO(0, serialized)
 ZEND_END_ARG_INFO()
 
+#if PHP_VERSION_ID >= 80100
```

Comment:
If you want to preserve full BC with PHP < 8.1, this looks right. If, on the other hand, that would not be necessary, you could do all the new PHP 8.1 stuff unconditionally. Prior to PHP 7.4.0, it would be ignored; as of PHP 7.4.0, `Serializable` would be ignored.

Reply:
Well, here I had many problems. I ended up with only breaking BC for 8.1 (I guess I am not really breaking anything since we were not really supporting 8.1).

1. If both `serialize()` and `__serialize()` are available PHP will pick the new one in 7.4+. Those two create totally different outputs, as seen in the tests I duplicated instead of changed because of this difference.
2. Method signature for `zend_call_method()` was changed in 8.0. So, I would have had to deal with that mismatch somehow.
3. This extension uses the unserialization process to create objects from the XML responses, so I don't really dare to change things too much :)